### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by MaxSchmide**

### CODE REVIEW

**Review by ollavka — APPROVED** [review]
Good job)

**Review by ollavka — APPROVED** [review]

**Review by YegorKorenkov — APPROVED** [review]
Looks good!

File: src/arrayMethodJoin.js [hunk 1]
```diff
@@ -4,8 +4,27 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = this[0] === null ? '' : `${this[0]}`;
+
+    const separate = separator === null ? 'null' : separator;
```
Comment (tetiana-ivanchenko) [inline src/arrayMethodJoin.js]: redundant

File: src/arrayMethodJoin.js [hunk 2]
```diff
@@ -4,8 +4,27 @@
...
+      return string;
+    }
+
+    for (let i = 1; i < this.length; i++) {
+      let element = this[i];
+
+      if (this[i] === null || this[i] === undefined) {
+        element = '';
+      }
+      string += separate + element;
```
Comment (tetiana-ivanchenko) [inline src/arrayMethodJoin.js]: no need to add empty string, check if element is not null and undefined, and add it only in this case

**Review by tetiana-ivanchenko — CHANGES_REQUESTED** [review]

File: src/arrayMethodJoin.js [hunk 3]
```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let resultString = '';
+
+    const separate = `${separator}`;
```
Comment (anstsot) [inline src/arrayMethodJoin.js]: no need to add additional value, just use `separator`

File: src/arrayMethodJoin.js [hunk 4]
```diff
@@ -4,8 +4,22 @@
...
+
+    for (let i = 0; i < this.length; i++) {
+      const element = this[i] === null || this[i] === undefined
+        ? ''
+        : this[i];
+
+      i === this.length - 1
+        ? resultString += element
+        : resultString += element + separate;
```
Comment (anstsot) [inline src/arrayMethodJoin.js]: try to rewrite by using `if ... else` construction or use the ternary operator after setting the value to `resultString`: `resultString += i === this.length - 1 ? ... : ....;`

**Review by anstsot — CHANGES_REQUESTED** [review]
Almost done :+1:
Let’s add a few fixes to improve your code.

**Review by anstsot — APPROVED** [review]
Good work, approving:sparkles:

**Pull request closed without merging**